pkg/testing/utils/types.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// Define some types that are only used for testing purposes
+
+type L4Endpoint struct {
```

Comment:
You could embed `L3Endpoint` here. The JSON would still stay all in one object and should work

Reply:
That's true, but the initialization would need something like:

L4Enpodint {
L3Endpoint {
...
...
}
}

Since it's only for testing purposes, I prefer to keep it that way.